[core] signing.passphrase must not be blank.

In a `signing {}` block like so:

```
    jreleaser {
        signing {
            setActive("ALWAYS")
            setMode("COMMAND")
        }
        ...
    }
```

I get the error:

```
[ERROR] signing.passphrase must not be blank.
```

But what if the PGP private key is not protected by a passphrase? It feels like jreleaser should not be enforcing it, should it?

--- a/api/jreleaser-resource-bundle/src/main/resources/org/jreleaser/bundle/Messages.properties
+++ b/api/jreleaser-resource-bundle/src/main/resources/org/jreleaser/bundle/Messages.properties
@@ -160,6 +160,7 @@ signing.file.newer                   = {} is newer than {}
 signing.check.published.key          = Checking if key {} has been published
 signing.key.published                = Key {} was found as published
 signing.key.not.published            = Key {} was NOT found as published
+signing.passphrase.blank             = Passphrase is blank. In case a passphrase is needed, you can configure a value using the {}, or define a System property {}, or define a {} environment variable, or define a key/value pair in {} with a key named {}
 signing.public.key.expiration.date   = Signing key {} expires at {}
 signing.public.key.no.expiration.date = Signing key {} does not expire
 ERROR_signing_verify_file            = Could not verify file {} with signature {}

--- a/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/signing/SigningValidator.java
+++ b/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/signing/SigningValidator.java
@@ -21,6 +21,7 @@ import org.jreleaser.bundle.RB;
 import org.jreleaser.model.api.JReleaserContext.Mode;
 import org.jreleaser.model.internal.JReleaserContext;
 import org.jreleaser.model.internal.deploy.maven.MavenDeployer;
+import org.jreleaser.model.internal.environment.Environment;
 import org.jreleaser.model.internal.signing.Signing;
 import org.jreleaser.util.DefaultVersions;
 import org.jreleaser.util.Errors;
@@ -38,6 +39,8 @@ import static org.jreleaser.model.api.signing.Signing.GPG_PUBLIC_KEYRING;
 import static org.jreleaser.model.api.signing.Signing.GPG_SECRET_KEY;
 import static org.jreleaser.model.internal.validation.common.Validator.checkProperty;
 import static org.jreleaser.model.internal.validation.common.Validator.resolveActivatable;
+import static org.jreleaser.util.Env.envKey;
+import static org.jreleaser.util.Env.sysKey;
 import static org.jreleaser.util.StringUtils.isBlank;
 
 /**
@@ -69,13 +72,26 @@ public final class SigningValidator {
 
         boolean cosign = signing.resolveMode() == org.jreleaser.model.Signing.Mode.COSIGN;
 
-        signing.setPassphrase(
-            checkProperty(context,
-                cosign ? COSIGN_PASSWORD : GPG_PASSPHRASE,
-                "signing.passphrase",
-                signing.getPassphrase(),
-                errors,
-                context.isDryrun()));
+        String passphrase = checkProperty(context,
+            cosign ? COSIGN_PASSWORD : GPG_PASSPHRASE,
+            "signing.passphrase",
+            signing.getPassphrase(),
+            new Errors(),
+            false);
+
+        if (isBlank(passphrase)) {
+            String key = cosign ? COSIGN_PASSWORD : GPG_PASSPHRASE;
+            Environment environment = context.getModel().getEnvironment();
+            String dsl = context.getConfigurer().toString();
+            String configFilePath = environment.getPropertiesFile().toAbsolutePath().normalize().toString();
+            String value = environment.resolve(key);
+            String envKey = envKey(key);
+            String sysKey = sysKey(key);
+            context.getLogger().warn(RB.$("signing.passphrase.blank", dsl, sysKey, envKey, configFilePath, key));
+            passphrase = "";
+        }
+
+        signing.setPassphrase(passphrase);
 
         if (signing.resolveMode() == org.jreleaser.model.Signing.Mode.COMMAND) {
             signing.getCommand().setExecutable(

